backend/logic/printful/sync/stockUpdate.js

```diff
@@ -0,0 +1,61 @@
+const get = require('lodash/get')
+const { decryptConfig } = require('../../../utils/encryptedConfig')
+
+const { Product } = require('../../../models')
+const { getLogger } = require('../../../utils/logger')
+
+const log = getLogger('utils.printful.writeProductData')
+
+/**
+ * Appends a productId to a collection.
+ * And creates collection if it doesn't exist
+ *
+ * @param {Object} shop
+ * @param {Number|String} productId
+ * @param {Array<Object>} variants
+ */
+const stockUpdate = async (shop, productId, variants) => {
+  try {
+    const shopConfig = decryptConfig(shop.config)
+    const shopInventory = get(shopConfig, 'inventory')
+    if (shopInventory) {
+      // Add to products model
+      const variantsStock = variants.reduce(
+        (stockData, variant) => ({
+          ...stockData,
+          [variant]: 0
+        }),
+        {}
+      )
+
+      const updatedStockData = {
+        productId: productId.toString(),
+        shopId: shop.id,
+        stockLeft: 0,
+        variantsStock
+      }
+
+      let existingDbItem
+
+      if (productId) {
+        existingDbItem = await Product.findOne({
+          where: {
+            productId: productId.toString(),
+            shopId: shop.id
+          }
+        })
+      }
+
+      if (existingDbItem) {
+        await existingDbItem.update(updatedStockData)
+      } else {
+        await Product.create(updatedStockData)
+      }
+    }
+  } catch (e) {
```

Comment:
Not sure we need a try/catch in this method since the caller already calls it within a try/catch.

Also, we very rarely check the system logs. So we'll probably not notice any error. It's good to leave that logging but in addition, it would probably make sense to log the error in Sentry since we have it configured to end us notifications by email in case of failure.

Reply:
Okay, I will remove it

Reply:
Removed